test/functional/pdo_sqlsrv/test_ae_keys_setup.phpt

```diff
@@ -1,36 +1,49 @@
---TEST--
-retrieval of names of column master key and column encryption key generated in the database setup
---SKIPIF--
-<?php require('skipif_unix.inc'); ?>
---FILE--
-<?php
-sqlsrv_configure( 'WarningsReturnAsErrors', 0 );
-sqlsrv_configure( 'LogSeverity', SQLSRV_LOG_SEVERITY_ALL );
-
-require( 'MsCommon.inc' );
-$conn = Connect();
-    
-if (IsAEQualified($conn)){
-    $query = "SELECT name FROM sys.column_master_keys";
-    $stmt = $conn->query($query);
-    $master_key_row = $stmt->fetch();
-    
-    $query = "SELECT name FROM sys.column_encryption_keys";
-    $stmt = $conn->query($query);
-    $encryption_key_row = $stmt->fetch();
-    
-    if ($master_key_row[0] == 'AEMasterKey' && $encryption_key_row[0] == 'AEColumnKey'){
-        echo "Test Successfully done.\n";
-    }
-    else {
-        die("Column Master Key and Column Encryption Key not created.\n");
-    }
-    unset($stmt);
-}
-else {
-    echo "Test Successfully done.\n";
-}
-unset($conn);
-?>
---EXPECT--
+--TEST--
+Test the existence of Windows Always Encrypted keys generated in the database setup
+--DESCRIPTION--
+This test iterates through the rows of sys.column_master_keys and/or 
+sys.column_encryption_keys to look for the specific column master key and 
+column encryption key generated in the database setup
+--SKIPIF--
+<?php require('skipif_unix.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon_mid-refactor.inc');
+$conn = connect();
+    
+if (isAEQualified($conn)){
+    $query = "SELECT name FROM sys.column_master_keys";
+    $stmt = $conn->query($query);
+
+    // Do not assume the master key must be the first one created
+    $found = false;
+    while ($master_key_row = $stmt->fetch()) {
+        if ($master_key_row[0] == 'AEMasterKey') {

```

Comment:
These key names are the defaults? Are these the names they get created with in our setup scripts?

Reply:
yes, by the AE setup scripts